soepy/python/simulate/simulate_auxiliary.py

```diff
@@ -18,36 +18,33 @@ def pyth_simulate(attr_dict, state_space_args, periods_emax):
         state_space_args[3],
     )
 
-    # Unpack parameter from the model specification
-    educ_min = attr_dict["INITIAL_CONDITIONS"]["educ_min"]
-    educ_max = attr_dict["INITIAL_CONDITIONS"]["educ_max"]
-    num_periods = attr_dict["GENERAL"]["num_periods"]
-    num_agents_sim = attr_dict["SIMULATION"]["num_agents_sim"]
-    seed_sim = attr_dict["SIMULATION"]["seed_sim"]
-    shocks_cov = attr_dict["DERIVED_ATTR"]["shocks_cov"]
-    optim_paras = attr_dict["PARAMETERS"]["optim_paras"]
-    delta = attr_dict["CONSTANTS"]["delta"]
-
-    educ_years = list(range(educ_min, educ_max + 1))
-    np.random.seed(seed_sim)
-    educ_years = np.random.choice(educ_years, num_agents_sim)
+    educ_years = list(range(model_params.educ_min, model_params.educ_max + 1))
+    np.random.seed(model_params.seed_sim)
+    educ_years = np.random.choice(educ_years, model_params.num_agents_sim)
 
     # Create draws for simulated sample
-    draws_sim = draw_disturbances((num_periods, num_agents_sim), shocks_cov, seed_sim)
+    draws_sim = draw_disturbances(
+        (model_params.num_periods, model_params.num_agents_sim),
```

Comment:
Why the use of a nested tuple in the interface here?

Reply:
The draw disturbances function is called during solve, simulate and test, every time with different arguments: different seed and more importantly different shape. Therefore it is much easier to supply the named tuple arguments directly. To have a common interface would produce some additional (and possibly unnecessary) lines of code.

Reply:
Yes, but my question refers to the use of the nested tuple in the inteface, `(model_params.num_periods, model_params.num_agents_sim), ...)` instead of `(model_params.num_periods, model_params.num_agents_sim, ...)`